src/Functions/formatReadable.h

```diff
@@ -78,10 +95,27 @@ class FunctionFormatReadable : public IFunction
     template <typename T>
     ColumnPtr executeType(const ColumnsWithTypeAndName & arguments) const
     {
+        const typename ColumnVector<Int8>::Container * precision = nullptr;
+        const typename ColumnVector<UInt8>::Container * precision2 = nullptr;
+        const ColumnVector<Int8> * precision_col = nullptr;
+        const ColumnConst * col_const = nullptr;
+        int precision_val = 2;
+        if (arguments.size() == 2)
+        {
+            precision_col = checkAndGetColumn<ColumnVector<Int8>>(arguments[1].column.get());
```

Comment:
In which cases it can be `Int8`? Normally it's a positive, so `UInt8` should be alright. 
Actually, I'd also support only constant there, since specifying different precision for different rows not really common usecase. Anyway, we can keep more generic implementation, but then please add tests for non-constant `precision`.

Reply:
I switched it to UInt8 and to always be constant, it makes more sense that way anyways. The other way made more sense until I started actually learning the differences between the column types and how IFunctions work.